pkg/v3/plugin/hooks/add_from_staging.go

```diff
@@ -35,18 +40,18 @@ func (hook *AddFromStagingHook) RunHook(obs *ocr2keepersv3.AutomationObservation
 	if err != nil {
 		return err
 	}
-
-	// Sort by a shuffled workID. workID for all items is shuffled using a pseduorandom souce
-	// that is the same across all nodes for a given round. This ensures that all nodes try to
-	// send the same subset of workIDs if they are available, while giving different priority
-	// to workIDs in different rounds.
-	sort.Slice(results, func(i, j int) bool {
-		return random.ShuffleString(results[i].WorkID, rSrc) < random.ShuffleString(results[j].WorkID, rSrc)
+	// creating a map to hold the shuffled workIDs
+	shuffledIDs := make(map[string]string, len(results))
+	for _, result := range results {
+		shuffledIDs[result.WorkID] = random.ShuffleString(result.WorkID, rSrc)
+	}
+	// sort by the shuffled workID
+	sort.SliceStable(results, func(i, j int) bool {
```

Comment:
Is SliceStable needed here?

Reply:
The issue with Slice is that it is not deterministic in case of duplicated items.
I can revert as we don't expect to have workID duplications, and to have this change cross all other places in code to be consistent